File: gherkinexecutor/Feature_Examples/SolutionForListOfNumber.py
from gherkinexecutor.Feature_Examples.ID import ID


class SolutionForListOfNumber:
    def __init__(self):
        self.values = []
        self.filter_value = ID("Q0000")

    def add(self, value):
        self.values.append(value)

    def set_filter_value(self, value):
        self.filter_value = value

    def sum(self):
        total = 0
        for element in self.values:
            if element.iD == self.filter_value:
                total += element.value
        return total

# Assuming LabelValueInternal and ID classes are defined elsewhere

File: gherkinexecutor/Feature_Examples/SolutionForListOfNumber.java
package gherkinexecutor.Feature_Examples;

import java.util.ArrayList;
import java.util.List;


public class SolutionForListOfNumber {
    private final List<LabelValueInternal> values = new ArrayList <>();
    private ID filterValue = new ID("Q0000");
    void add(LabelValueInternal value) {
        values.add(value);
    }

    void setFilterValue(ID value) {
        filterValue = value;
    }

    int sum(){
        var sum = 0;
        for (LabelValueInternal element : values) {
            if (element.iD.equals(filterValue))
                sum += (element.value);
        }
        return sum;
    }
}
